src/algorithms/peter_norving.py

```diff
@@ -0,0 +1,153 @@
+        grid -- grid skeleton
+        return grid dict with the filled values ready to play.
+        """
+
+        chars = [colum for colum in grid if colum in self.digits
+                 or colum in '0']
+        assert len(chars) == 81
+        return dict(zip(self.squares, chars))
+
+    def assign(self, values, square, digit):
+        """
+        Eliminate all the other values (except the sent digit)
+        from values[square] and then propagate.
+
+        Keyword arguments:
+        values -- initial dict with values
+        square -- square positions
+        digit --  selected digit
+        return updated values, except return False if a contradiction is detected.
+        """
+
+        other_values = values[square].replace(digit, '')
+        if all(self.eliminate(values, square, temp_digit)
+               for temp_digit in other_values):
+            return values
+        else:
+            return False
+
+    def eliminate(self, values, square, digit):
+        """
+        Eliminate digit from values[square]; propagate when values or places <= 2.
+
+        Keyword arguments:
+        values -- initial dic
+        square -- square positions
+        digit -- digit values
+        return updated values, except return False if a contradiction is detected.
+        """
+
+        if digit not in values[square]:
+            return values
+        values[square] = values[square].replace(digit, '')
+        if len(values[square]) == 0:
+            return False
+        elif len(values[square]) == 1:
+            temp_digit = values[square]
+            if not all(self.eliminate(values, temp_square, temp_digit)
+                       for temp_square in self.peers[square]):
+                return False
+        for unit in self.units[square]:
+            dplaces = [square for square in unit if digit
+                       in values[square]]
+            if len(dplaces) == 0:
+                return False
+            elif len(dplaces) == 1 and not self.assign(values,
+                    dplaces[0], digit):
+                return False
+        return values
+
+    def search(self, values):
+        """
+        Using depth-first search and propagation, try all possible values.
+
+        Keyword arguments:
+        values -- updates possible values dict.
+        return updated dict with possible values after search algorithm.
+        """
+
+        if values is False:
+            return False
+        if all(len(values[square]) == 1 for square in self.squares):
+            return values
+        (unfilled, square) = min((len(values[square]), square)
+                                 for square in self.squares
+                                 if len(values[square]) > 1)
+
+        return self.get_true_element(self.search(self.assign(values.copy(),
+                square, digit)) for digit in values[square])
+
+    def get_true_element(self, sequence):
+        """
+        return the element that is true.
```

Comment:
The description of the method is saying the same that the name, and this is not understandable, when an element is true? what is an element?